.. change::
    :tags: bug, orm
    :tickets: 5034

    Fixed issue where by if the "begin" of a transaction failed at the Core
    engine/connection level, such as due to network error or database is locked
    for some transactional recipes, within the context of the :class:`.Session`
    procuring that connection from the conneciton pool and then immediately
    returning it, the ORM :class:`.Session` would not close the connection
    despite this connection not being stored within the state of that
    :class:`.Session`.  This would lead to the connection being cleaned out by
    the connection pool weakref handler within garbage collection which is an
    unpreferred codepath that in some special configurations can emit errors in
    standard error.
